flink-connector/src/main/java/com/datastrato/gravitino/flink/connector/store/GravitinoCatalogStore.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.flink.connector.store;
+
+import com.datastrato.gravitino.Catalog;
+import com.datastrato.gravitino.flink.connector.PropertiesConverter;
+import com.datastrato.gravitino.flink.connector.catalog.GravitinoCatalogManager;
+import com.datastrato.gravitino.flink.connector.hive.GravitinoHiveCatalogFactory;
+import com.datastrato.gravitino.flink.connector.hive.HivePropertiesConverter;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.table.catalog.AbstractCatalogStore;
+import org.apache.flink.table.catalog.CatalogDescriptor;
+import org.apache.flink.table.catalog.CommonCatalogOptions;
+import org.apache.flink.table.catalog.exceptions.CatalogException;
+import org.apache.flink.util.Preconditions;
+
+/** GravitinoCatalogStore is used to store catalog information to Gravitino server. */
+public class GravitinoCatalogStore extends AbstractCatalogStore {
+
+  private final GravitinoCatalogManager gravitinoCatalogManager;
+
+  public GravitinoCatalogStore(GravitinoCatalogManager catalogManager) {
+    this.gravitinoCatalogManager = catalogManager;
+  }
+
+  @Override
+  public void storeCatalog(String catalogName, CatalogDescriptor descriptor)
+      throws CatalogException {
+    Configuration configuration = descriptor.getConfiguration();
+    String provider = getGravitinoCatalogProvider(configuration);
+    Catalog.Type type = getGravitinoCatalogType(configuration);
+    Map<String, String> gravitinoProperties =
+        getPropertiesConverter(provider).toGravitinoCatalogProperties(configuration);
+    gravitinoCatalogManager.createCatalog(catalogName, type, "", provider, gravitinoProperties);
+  }
+
+  @Override
+  public void removeCatalog(String catalogName, boolean ignoreIfNotExists) throws CatalogException {
+    try {
+      gravitinoCatalogManager.dropCatalog(catalogName);
+    } catch (Exception e) {
+      throw new CatalogException(e);
+    }
+  }
+
+  @Override
+  public Optional<CatalogDescriptor> getCatalog(String catalogName) throws CatalogException {
+    try {
+      Catalog catalog = gravitinoCatalogManager.getGravitinoCatalogInfo(catalogName);
+      String provider = catalog.provider();
+      PropertiesConverter propertiesConverter = getPropertiesConverter(provider);
+      Map<String, String> flinkCatalogProperties =
+          propertiesConverter.toFlinkCatalogProperties(catalog.properties());
+      CatalogDescriptor descriptor =
+          CatalogDescriptor.of(catalogName, Configuration.fromMap(flinkCatalogProperties));
+      return Optional.of(descriptor);
+    } catch (Exception e) {
```

Comment:
please add log here

Reply:
Done

Reply:
would you like to use `LOG` to keep consistent with other modules?